include/dxc/DxilContainer/DxilPipelineStateValidation.h

```diff
@@ -432,133 +563,138 @@ class DxilPipelineStateValidation
+          PSV_RETB(rw.MapArray(&m_pSigPatchConstOrPrimElements, m_pPSVRuntimeInfo1->SigPatchConstOrPrimElements, m_uPSVSignatureElementSize));
+        }
       }
 
       // ViewID dependencies
       if (m_pPSVRuntimeInfo1->UsesViewID) {
         for (unsigned i = 0; i < 4; i++) {
           if (m_pPSVRuntimeInfo1->SigOutputVectors[i]) {
-            minsize += sizeof(uint32_t) * PSVComputeMaskDwordsFromVectors(m_pPSVRuntimeInfo1->SigOutputVectors[i]);
-            if (!(size >= minsize)) return false;
-            m_pViewIDOutputMask = (uint32_t*)pCurBits;
-            pCurBits += sizeof(uint32_t) * PSVComputeMaskDwordsFromVectors(m_pPSVRuntimeInfo1->SigOutputVectors[i]);
+            PSV_RETB(rw.MapArray(&m_pViewIDOutputMask,
+              PSVComputeMaskDwordsFromVectors(m_pPSVRuntimeInfo1->SigOutputVectors[i])));
           }
           if (!IsGS())
             break;
         }
         if ((IsHS() || IsMS()) && m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors) {
-          minsize += sizeof(uint32_t) * PSVComputeMaskDwordsFromVectors(m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors);
-          if (!(size >= minsize)) return false;
-          m_pViewIDPCOrPrimOutputMask = (uint32_t*)pCurBits;
-          pCurBits += sizeof(uint32_t) * PSVComputeMaskDwordsFromVectors(m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors);
+          PSV_RETB(rw.MapArray(&m_pViewIDPCOrPrimOutputMask,
+            PSVComputeMaskDwordsFromVectors(m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors)));
         }
       }
 
       // Input to Output dependencies
       for (unsigned i = 0; i < 4; i++) {
         if (!IsMS() && m_pPSVRuntimeInfo1->SigOutputVectors[i] > 0 && m_pPSVRuntimeInfo1->SigInputVectors > 0) {
-          minsize += PSVComputeInputOutputTableSize(m_pPSVRuntimeInfo1->SigInputVectors, m_pPSVRuntimeInfo1->SigOutputVectors[i]);
-          if (!(size >= minsize)) return false;
-          m_pInputToOutputTable = (uint32_t*)pCurBits;
-          pCurBits += PSVComputeInputOutputTableSize(m_pPSVRuntimeInfo1->SigInputVectors, m_pPSVRuntimeInfo1->SigOutputVectors[i]);
+          PSV_RETB(rw.MapArray(&m_pInputToOutputTable,
+            PSVComputeInputOutputTableDwords(m_pPSVRuntimeInfo1->SigInputVectors, m_pPSVRuntimeInfo1->SigOutputVectors[i])));
         }
         if (!IsGS())
           break;
       }
-      if ((IsHS() || IsMS()) && m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors > 0 && m_pPSVRuntimeInfo1->SigInputVectors > 0) {
-        minsize += PSVComputeInputOutputTableSize(m_pPSVRuntimeInfo1->SigInputVectors, m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors);
-        if (!(size >= minsize)) return false;
-        m_pInputToPCOutputTable = (uint32_t*)pCurBits;
-        pCurBits += PSVComputeInputOutputTableSize(m_pPSVRuntimeInfo1->SigInputVectors, m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors);
+      if (IsHS() && m_pPSVRuntimeInfo1->SigPatchConstOrPrimVectors > 0 && m_pPSVRuntimeInfo1->SigInputVectors > 0) {
```

Comment:
IsMS() was removed intentionally?

Reply:
It was misleading and shouldn't have been there...
It actually can't participate because MS can't have nonzero SigInputVectors, and shouldn't ever have this info.  It doesn't actually change behavior, and previous code paths for sizing and writing did not include MS, so it was a mistake to be included here.